src/search.rs

```diff
@@ -666,6 +730,29 @@ mod tests {
         kind: String,
         number: i32,
         nested: Nested,
+        #[serde(default)]
+        _vectors: Option<Vectors>,
+    }
+
+    #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    struct Vector {
+        embeddings: Vec<Vec<f32>>,
```

Comment:
Here meilisearch can return a `Vec<f32>` directly.

You could do something like that;
```rust
    #[derive(Debug, Serialize, Deserialize, PartialEq)]
    struct Vector {
        embeddings: SingleOrMultipleVectors,
        regenerate: bool,
    }

    #[derive(Serialize, Deserialize)]
    #[serde(untagged)]
    enum SingleOrMultileVectors {
        Single(Vec<f32>),
        Multiple(Vec<Vec<f32>>),
    }
```